lib/msf/core/exploit/remote/http_client.rb

```diff
@@ -373,27 +374,97 @@ def send_request_raw(opts = {}, timeout = 20, disconnect = false)
   # Connects to the server, creates a request, sends the request,
   # reads the response
   #
+  # If a +HTTP::CookieJar+ instance is passed in the +opts+ dict under a 'cookie' key, said CookieJar will be used in
+  # the request instead of the module +cookie_jar+
+  #
   # Passes `opts` through directly to {Rex::Proto::Http::Client#request_cgi}.
   # Set `opts['keep_cookies']` to keep cookies from responses for reuse in requests.
+  # Cookies returned by the server will be stored in +cookie_jar+
   #
   # @return (see Rex::Proto::Http::Client#send_recv))
   def send_request_cgi(opts = {}, timeout = 20, disconnect = true)
-    if cookie_jar.any?
-      opts = { 'cookie' => cookie_jar.to_a.join(' ') }.merge(opts)
+    url = "http#{ssl ? 's' : ''}://#{peer}"
+
+    if opts.has_key?('cookie') && opts['cookie'].class == HTTP::CookieJar
```

Comment:
I don't know if it really matters here but you could use `instance_of?` here or `is_a?` I think they're all functionally identical though but might read better

Reply:
Would be more readable, will add it in